src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -555,4 +572,77 @@ private String getHashOfComponentName(String componentName) throws HashingAlgori
                     "Failed to compute filename because desired hashing algorithm is not available", e);
         }
     }
+
+    /**
+     * issues #1111
+     * Get all locally available component recipe content
+     *
+     * @return list of all component recipe content
+     */
+    public List<String> getAllComponentRecipeContent() {
```

Comment:
we should not be reading all the component recipes into memory, this will use too much memory.

Let's target only what we actually need. Get a list of the components, then for each component we want to read the artifacts and determine which docker image(s) would be safe to delete.